### PR TITLE
[PIN-1367] Migrating purpose-process POST /purposes/.../update/draft to BFF

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -4011,6 +4011,110 @@ paths:
...
+                type: integer
+              description: Remaining requests within time interval
+            'X-Rate-Limit-Interval':
+              schema:
+                type: integer
+              description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PurposeVersion'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: does this endpoint require to return the whole object or just the id?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala [hunk 2]
```diff
@@ -112,4 +112,27 @@ final case class PurposesApiServiceImpl(
...
+  ): Route = {
+    val result: Future[PurposeVersion] = for {
+      purposeUUID                <- purposeId.toFutureUUID
+      versionUUID                <- versionId.toFutureUUID
+      updatedDraftPurposeVersion <- purposeProcessService
+        .updateDraftPurposeVersion(purposeUUID, versionUUID, draftPurposeVersionUpdateContent.toProcess)
+    } yield updatedDraftPurposeVersion.toApi
+
+    onComplete(result) {
+      handleError(s"Error updating purpose's version in draft with Id: $purposeId") orElse { case Success(response) =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]:
```suggestion
      handleError(s"Error updating draft version $versionId of purpose $purposeId") orElse { case Success(response) =>
```

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PurposeProcessServiceImpl.scala [hunk 3]
```diff
@@ -47,4 +52,20 @@ class PurposeProcessServiceImpl(purposeProcessUrl: String, blockingEc: Execution
...
+    (bearerToken, correlationId, ip) =>
+      val request: ApiRequest[PurposeVersion] =
+        api.updateDraftPurposeVersion(
+          xCorrelationId = correlationId,
+          purposeId = purposeId,
+          versionId = versionId,
+          draftPurposeVersionUpdateContent = updateContent,
+          xForwardedFor = ip
+        )(BearerToken(bearerToken))
+      invoker.invoke(request, s"Updating purpose's version in draft")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PurposeProcessServiceImpl.scala]:
```suggestion
      invoker.invoke(request, s"Updating draft version $versionId of purpose $purposeId")
```

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/PurposeProcessServiceTypes.scala [hunk 4]
```diff
@@ -5,6 +5,10 @@ import it.pagopa.interop.purposeprocess.client.{model => PurposeProcess}
 
 object PurposeProcessServiceTypes {
 
+  type DraftPurposeVersionProcess    = PurposeProcess.DraftPurposeVersionUpdateContent
+  type PurposeVersionProcess         = PurposeProcess.PurposeVersion
+  type PurposeVersionDocumentProcess = PurposeProcess.PurposeVersionDocument
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/types/PurposeProcessServiceTypes.scala]: are these used? if not, you can remove them

**Review by galales — APPROVED** [review]

**Pull request merged**